### PR TITLE
[DOC] Speed up fit computation with parallel processing

### LINKED ISSUE #3980: Examples with long build time
https://nilearn.github.io/dev/auto_examples/02_decoding/plot_haxby_frem.html (10m59s) (locally 6m29s)
https://nilearn.github.io/dev/auto_examples/03_connectivity/plot_compare_decomposition.html (7m49s)

extracted from https://github.com/nilearn/nilearn/issues/3728#issue-1686548159

### PR DESCRIPTION
**Pull request by ymzayek**
- Closes #3980

### CODE REVIEW

**Comment by bthirion** [comment]
Can you assess the impact on build time ?

**Comment by ymzayek** [comment]
In the CI, [plot_haxby_frem.html](https://nilearn.github.io/dev/auto_examples/02_decoding/plot_haxby_frem.html) now builds in a bit over 2 minutes and for [plot_compare_decomposition.html](https://nilearn.github.io/dev/auto_examples/03_connectivity/plot_compare_decomposition.html) it's about 4min. This time will depend on whether data is downloaded or cache is used. For this reference build cache is not used. Locally this change shaves off about 2-3 minutes for each example.

**Review by Remi-Gau — APPROVED** [review]
Good with me.
Out of curiosity: how many cpus can we use in github CI?

**Comment by ymzayek** [comment]
@Remi-Gau in general 2 : https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

**Comment by bthirion** [comment]
Are there other examples that  could benefit from this to further reduce doc generation time ? If yes, we should do it in this PR.

**Comment by ymzayek** [comment]
Good point. I'll check and update

**Comment by ymzayek** [comment]
I set the examples to use max cpu available with -1. It seems each CPU is dual-core so we end up with 4 processes

**Comment by Remi-Gau** [comment]
that's luxurious! 💲

**Comment by bthirion** [comment]
Please do not leave `n_jobs=-1` in the examples.

**Comment by ymzayek** [comment]
@bthirion yes I thought about that but wanted to check the limit. Probably safest to leave n_jobs=2 so as to not have people run into memory issues when running the examples as they are

File: examples/02_decoding/plot_haxby_searchlight.py [hunk 1]
```diff
@@ -67,7 +67,7 @@
 # Make processing parallel
 # /!\ As each thread will print its progress, n_jobs > 1 could mess up the
 #     information output.
-n_jobs = 1
+n_jobs = -1
```
Comment (ymzayek) [inline examples/02_decoding/plot_haxby_searchlight.py:70]:
```suggestion
n_jobs = 2
```

**Comment by ymzayek** [comment]
The build with n_jobs=2 (https://github.com/nilearn/nilearn/actions/runs/6892690674/job/18750572159) now takes ~1h10m which is quite a speed up. This is with using data cache.

**Comment by bthirion** [comment]
Should we apply that to other examples too ?

**Comment by ymzayek** [comment]
@bthirion it's possible I've missed a few. I'll double check

**Review by bthirion — APPROVED** [review]
LGTM !

**Comment by ymzayek** [comment]
Thanks merging

**Pull request merged**